src/uu/shuf/src/shuf.rs

```diff
@@ -59,13 +62,19 @@ pub fn uumain(args: impl uucore::Args) -> UResult<()> {
             }
         }
     } else {
-        Mode::Default(
-            matches
-                .get_one::<String>(options::FILE)
-                .map(|s| s.as_str())
-                .unwrap_or("-")
-                .to_string(),
-        )
+        match matches.get_many::<String>(options::FILE_OR_ARGS) {
+            Some(mut args) if args.len() == 1 => Mode::Default(args.next().unwrap().to_string()),
+            Some(mut args) => {
+                debug_assert!(args.len() >= 2);
+                args.next().unwrap();
+                let second_file = args.next().unwrap();
+                return Err(UUsageError::new(
+                    1,
+                    format!("unexpected argument '{second_file}' found"),
+                ));
+            }
+            None => Mode::Default("-".to_string()),
+        }
```

Comment:
I've tried to reduce the branching and unwraps a bit. Let me know what you think, it's a tricky bit of code.
```suggestion
        let mut operands = matches
            .get_many::<String>(options::FILE_OR_ARGS)
            .unwrap_or_default();

        let file = operands
            .next()
            .map(Clone::clone)
            .unwrap_or("-".into())
            .to_string();

        if let Some(second_file) = operands.next() {
            return Err(UUsageError::new(
                1,
                format!("unexpected argument '{second_file}' found"),
            ));
        };

        Mode::Default(file)
```